### PR TITLE
Redesign the all domains list component

### PR DESCRIPTION
**Pull request by hambai**
#### Changes proposed in this Pull Request

* Add new `AllDomains` component that'll render the new all domains list redesigned view
* Fixed the ordering by Site in the all domains list
* Added filter dropdown in the all domains list
* Refactor the `DomainRow` component to remove duplicate `domainDetails` prop
* Add Site column in the `DomainRow` component when rendering the all domains view
* Fix how the `selectedSite` is passed down to the `DomainsTable` component
* In order to make this work I'm loading all the domain data initially - it might be slow but all the filtering and sorting should work

What's done and what's left that I'll fix in a follow up PRs: 
- [ ] add loading state for the filter dropdown
- [ ] add loading state for the `DomainRow` component
- [ ] fix the auto-renew toggle but (sometimes it stays in loading state forever)
- [ ] add domain ownership value in the REST API that we should use for the domain filter
- [ ] fix the bulk contact editing
- [ ] show the filter dropdown on mobile
- [x] lazy load the detailed domain data - we'll load all the domain data for now (we won't use the lazy loading 
- [x] render the site as link to the site domains list (and how to render domain-only sites)
- [x] make the sorting by site column work
- [x] render the site in mobile view
- [x] refactor the `domain`/`domainDetails` prop in the DomainRow component (we don't need both)
- [x] add the filter button

It should look like this:

Screenshot 2021-11-12 at 14 16 28
Screenshot 2021-11-12 at 14 16 53

#### Testing instructions

* Spin up this branch or calypso.live and go to /domains/manage . Verify that the all domains list renders as expected on desktop and mobile and that the sorting works
* Some of the functionality might not work as expected (see the above list)

### CODE REVIEW

File: client/my-sites/domains/domain-management/list/all-domains.jsx [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+		);
+	}
+
+	renderBreadcrumbs() {
+		const { translate } = this.props;
+
+		const item = {
+			label: translate( 'All Domains' ),
+			helpBubble: translate(
+				'Manage all your domains. {{learnMoreLink}}Learn more{{/learnMoreLink}}.',
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/list/all-domains.jsx:667]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Manage the domains connected to your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.' )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/all-domains-list-redesign)

File: client/my-sites/domains/domain-management/list/domain-row.jsx [hunk 2]
```diff
@@ -30,17 +32,16 @@ class DomainRow extends PureComponent {
 		currentRoute: PropTypes.string,
 		disabled: PropTypes.bool,
 		domain: PropTypes.object.isRequired,
-		domainDetails: PropTypes.object,
```
Comment (hambai) [inline client/my-sites/domains/domain-management/list/domain-row.jsx:33]: Getting rid of the `domainDetails` property as I'm merging all the data in the DomainsTable component

**Review by delputnam — APPROVED** [review]
This LGTM and works as expected with one exception related to sorting in desc order by the "registered until" field.  ...which I think is related to some of the functionality that isn't ready yet.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6978903

Thank you @hambai for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6978903).

**Pull request merged**